frontend/src/utils/index.js

```diff
@@ -571,22 +571,33 @@ export function generateWorker (availableZones, cloudProfileName, region) {
     maximum: 2,
     maxSurge: 1,
     machine: {
-      type: machineType,
+      type: machineType.name,
       image: machineImage
     },
     zones,
     isNew: true
   }
-  if (volumeType) {
+  if (volumeType.name) {
+    worker.volume = {
+      type: volumeType.name,
+      size: defaultVolumeSize
+    }
+  } else if (machineType.storage && !machineTypeHasStorageWithTypeFixed(machineType)) {
+    worker.volume = {
+      size: machineType.storage.size
+    }
+  } else if (!machineType.storage) {
     worker.volume = {
-      type: volumeType,
       size: defaultVolumeSize
     }
   }
-
   return worker
 }
 
+export function machineTypeHasStorageWithTypeFixed (machineType) {
```

Comment:
better name maybe: isMachineTypeStorageFixed ?

Reply:
Agree

Reply:
We should change the helper method to `isMachineTypeStorageNotFixed` and also check machineType.store

Reply:
renamed to `machineTypeHasStorageButNotWithTypeFixed`

Reply:
and added check for machineType.storage